src/main/scala/zio/zmx/diagnostics/parser/Parser.scala

```diff
@@ -0,0 +1,93 @@
+        /* Parsed to RESP value */
+        (result: Resp.RespValue) => {
+
+          /*
+           * Commands must be sent in RESP format as Array of Bulk Strings
+           * First element is obligatory and will be the command name
+           * Consecutive elements are optional and will be arguments of the command
+           */
+          val invalidRequest =
+            IO.fail(ZMXProtocol.Error.InvalidRequest("Expected Array of Bulk Strings with at least one element"))
+
+          result match {
+            case Resp.Array(items) =>
+              /* Collect Bulk Strings only */
+              val bulkStrings: Chunk[Resp.BulkString] = items.collect { case bulkString: Resp.BulkString =>
+                bulkString
+              }
+
+              if (bulkStrings.size != items.size) {
+                /* Not all items in Array were Bulk Strings */
+                invalidRequest
+              } else {
+                bulkStrings match {
+                  case head +: tail =>
+                    val commandName = head.value
+                    val arguments   = tail.map(_.value)
+
+                    ZMXProtocol.Command
+                      .fromString(commandName)
+                      .fold[IO[ZMXProtocol.Error, ZMXProtocol.Request]] {
+                        /* Unknown command */
+                        IO.fail(ZMXProtocol.Error.UnknownCommand(commandName))
+                      } { command =>
+                        /* Correct command with optional arguments */
+                        IO.succeed {
+                          ZMXProtocol.Request(
+                            command,
+                            arguments.nonEmptyOrElse[Option[Chunk[String]]] {
+                              None
+                            } {
+                              Some(_)
+                            }
+                          )
+                        }
+                      }
+
+                  case _ => invalidRequest
+                }
+              }
+
+            case _ => invalidRequest
+          }
+
+        }
+      )
+
+  def serialize(response: ZMXProtocol.Response): Chunk[Byte] = response match {
+    case ZMXProtocol.Response.Fail(error) =>
+      error match {
+        case ZMXProtocol.Error.InvalidRequest(error)   => Resp.Error(s"INVALID REQUEST: `$error`!").serialize
+        case ZMXProtocol.Error.MalformedRequest(error) => Resp.Error(s"MALFORMED REQUEST: `$error`!").serialize
+        case ZMXProtocol.Error.UnknownCommand(command) => Resp.Error(s"UNKNOWN COMMAND: `$command`!").serialize
+      }
+
+    case ZMXProtocol.Response.Success(data) =>
+      data match {
+        case executionMetrics: ZMXProtocol.Data.ExecutionMetrics =>
+          // TODO: Format of `ExecutionMetrics` serialized could be discussed and revisited.
+          Resp.BulkString(executionMetrics.toString).serialize
+
```

Comment:
Regarding this, I will create an issue soon.
I think we can improve current format with recently added proper RESP implementation.

Right now whole metrics are a single Bulk String, eg.:
```
$86\r\n
concurrency:1\r\n
capacity:2\r\n
size:3\r\n
enqueued_count:4\r\n
dequeued_count:5\r\n
workers_count:6\r\n
```

Didn't want to break the current approach in this PR but I think it makes sense to use an Array here.
Actually the `key:value` pairs can be Arrays themselves as well.